Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -211,8 +252,16 @@ private void ProcessPointer(ref PointerModel state)
             }
             else if (pointerType == UIPointerType.Tracked)
             {
-                eventData.trackedDeviceOrientation = state.worldOrientation;
-                eventData.trackedDevicePosition = state.worldPosition;
+                var position = state.worldPosition;
+                var rotation = state.worldOrientation;
+                if (m_XRTrackingOrigin != null)
+                {
+                    position = m_XRTrackingOrigin.TransformPoint(position);
+                    rotation = m_XRTrackingOrigin.rotation * rotation;
```

Comment:
this is wrong, you need `TransformDirection`, otherwise if parent of `m_XRTrackingOrigin` has rotation, the result will be incorrect

Reply:
Can you elaborate?  This should transform the InputAction rotation into the tracking origin space.
I made a quick script test:
```
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class TransformTest : MonoBehaviour
{
    // Base XR Stage
    [SerializeField]
    Transform m_XrTransform;

    // Transform used as a real world reference (should be parented to m_XrTransform)
    [SerializeField]
    Transform m_ReferenceLocalTransform;

    // World space Transform that should be the same global position & rotation as ReferenceLocalTransform
    [SerializeField]
    Transform m_TargetTransform;
    

    // Update is called once per frame
    void LateUpdate()
    {
        var localPosition = m_ReferenceLocalTransform.localPosition;
        var localRotation = m_ReferenceLocalTransform.localRotation;

        m_TargetTransform.position = m_XrTransform.TransformPoint(localPosition);
        m_TargetTransform.rotation = m_XrTransform.rotation * localRotation;
    }
}

```

And when I move and rotate the XRTransform or ReferenceLocalTransform game objects, the target tracks it consistently.
I feel like using TransformDirection would be more steps for the same results.

Reply:
`m_XrTransform` is a hierarchy of transforms, e.g. it has `m_XrTransform.parent`
Does `m_XrTransform.transform` include parent rotation as well?

Reply:
Ah right! Deeper Hierarchies.
Yeah, I want to include all parent transform rotations as well. I'm using .rotation instead of .localRotation.
Tested my little script with an extra layer of GameObject on top of XrTransform and it still places the object where I'd expect it to be.